### PR TITLE
adam: Remove reference to adam-cli/target in head build

### PR DESCRIPTION
**Pull request by heuermh**
Due to commit https://github.com/bigdatagenomics/adam/commit/51b755b9d9eb105dbc9a960eb68ce2ca01ab486c the build has changed for `--HEAD` builds.

### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [x] Built your formula locally prior to submission with `brew install <formula>`?

### Updates to existing formula: have you

- [x] Removed the `revision` line, if any, when bumping the version number?
- [x] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [x] Not altered the `bottle` section?
- [x] Checked that the tests still pass?

### CODE REVIEW

**Comment by heuermh** [comment]
Separate commit to address brew audit errors, will squash before merge if desired.

**Comment by heuermh** [comment]
Rebased to fix conflicts and squashed audit related commit which made changes similar to 4e7e076a1ef1144aa5b5ae509520950964d36422 by @apjanke.

We're in a bit of a fix though, while HEAD is fine as the Travis build shows our release version 0.19.0 is incompatible with Spark 2.0.  Does that mean we can't make this fix for HEAD until we also cut a 0.19.1 release?

File: adam.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class Adam < Formula
-  desc "Genomics analysis platform using Apache Avro, Spark, and Parquet"
+  desc "Genomics analysis platform built on Apache Avro, Apache Spark and Parquet"
   homepage "https://github.com/bigdatagenomics/adam"
   url "https://search.maven.org/remotecontent?filepath=org/bdgenomics/adam/adam-distribution_2.10/0.19.0/adam-distribution_2.10-0.19.0-bin.tar.gz"
   sha256 "9010b198e8fc38de24030a2d00ae86351a6a3af2a3f62b3d21251648e3d60524"
```
Comment (dpo) [inline adam.rb]: Needs a `revision`.

Comment (heuermh) [inline adam.rb]: Does it?  The version number and precompiled bottles should be the same, the changes are only intended for the HEAD build, which builds from source.

Comment (dpo) [inline adam.rb]: You're right, sorry.

**Comment by dpo** [comment]
@BrewTestBot test this please

**Comment by heuermh** [comment]
Closing in favor of #4510

**Pull request closed without merging**